kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -226,7 +220,8 @@ private static <T> void putOrMerged(ManagedWorkflowAndDependentResourceContext c
         }
     }
 
-    private SecureConfigInterpolator.InterpolationResult interpolateConfig(KafkaProtocolFilterSpec spec) {
+    private SecureConfigInterpolator.InterpolationResult interpolateConfig(Context<KafkaProxy> context, KafkaProtocolFilterSpec spec) {
+        var secureConfigInterpolator = context.managedWorkflowAndDependentResourceContext().getMandatory("sec", SecureConfigInterpolator.class);
```

Comment:
go via the `SharedKafkaProxyContext` class to share/hide the key

Reply:
Aside: I now think that `SharedKafkaProxyContext` was probably a mistake. I don't think we want to end up with a `Shared*Context` class for each Reconciler.  The `Context` is associated with the `Reconciler`, so I think we should move those static methods from `SharedKafkaProxyContext` to the `KafkaProxyReconciler`.

Reply:
I started following this pattern. I will open another PR to move those methods and kill `SharedKafkaProxyContext`.

Reply:
#1975